tests/test_solvers.py

```diff
@@ -0,0 +1,295 @@
+                    'size',
+                    'subdir',
+                    'timestamp',
+                    'version',
+                    'track_features',
+                    'features',
+                )
+            }
+        # write repodata
+        for subdir in subdirs:
+            subdir_path = channel_path / subdir
+            subdir_path.mkdir(parents=True, exist_ok=True)
+            subdir_path.joinpath('repodata.json').write_text(json.dumps({
+                'info': {
+                    'subdir': subdir,
+                },
+                'packages': package_data.get(subdir, {}),
+            }))
+
+    @contextlib.contextmanager
+    def env(self, key, value):
+        old = os.environ.get(key)
+        os.environ[key] = value
+        try:
+            yield
+        except:
+            if old is not None:
+                os.environ[key] = old
+
+    @contextlib.contextmanager
+    def solver(self, *, add=(), remove=(), packages=()):
+        """Create a new solver.
+
+        Roughly equivalent to ``Solver(specs_to_add=..., specs_to_remove=...)``
+        with a custom channel containing the data specified in `packages`.
+        """
+        with tempfile.TemporaryDirectory(prefix='conda-test-repo-') as tmpdir_path:
+            tmpdir = pathlib.Path(tmpdir_path)
+            channels = tuple(
+                Channel(f'file://{tmpdir / subdir}')
+                for subdir in context.subdirs
+            )
+            self._write_channel_packages(tmpdir, context.subdirs, list(packages))
+            yield self.solver_class(
+                prefix='dummy - does not exist',
+                subdirs=context.subdirs,
+                channels=channels,
+                specs_to_add=add,
+                specs_to_remove=remove,
+            )
+
+    def install(self, *specs, packages=None):
+        """Try a install transaction in channel with a set of specified packages."""
+        if not packages:
+            packages = self.index(1)
+        with self.solver(add=specs, packages=packages) as solver:
+            return solver.solve_final_state()
+
+    def index(self, num):
+        """Get the index data of the ``helpers.get_index_r_*`` helpers."""
+        get_index = getattr(helpers, f'get_index_r_{num}')
+        index, _ = get_index(context.subdir)
+        return index.values()
+
+    def package_string_set(self, packages):
+        """Transforms package container in package string set."""
+        return {
+            f'{record.name}-{record.version}-{record.build}'
+            for record in packages
+        }
+
+    def assert_installed(self, specs, expecting):
+        """Helper to assert that a transaction result contains the packages
+        specified by the set of specification string."""
+        installed = self.install(*specs)
+        assert installed, f'no installed specs ({installed})'
+        assert self.package_string_set(installed) == set(expecting) | {
+            # XXX: injected
```

Comment:
Huh, where are the distribute and pip specs come from?

Reply:
It's an option in the Conda settings, `CONDA_ADD_PIP_AS_PYTHON_DEPENDENCY` or something like that.

Reply:
I tried setting that env var, but it didn't have any effect. Maybe it gets evaluated at import time?

Reply:
I think they get evaluated upon access time; i.e. `context.add_pip_as_python_dependency` will check whether the env var is defined, return that or, otherwise, return the value set initially (via condarc or cli).

Reply:
I think what I'm wondering is whether "distribute" is really needed here, since that's an ancient package.

Reply:
The tests are also old (py27) so maybe the repodata was frozen a while ago?